### PR TITLE
WIP: Implement dictionary pointer and variable

### PR DESCRIPTION
**Pull request by Lapixx**

### CODE REVIEW

File: lib/core.fs [hunk 1]
```diff
@@ -406,6 +406,25 @@ ps-drop,
 ret,
 end-code
 
+( -- addr )
+code here
+ps-dup,
+$C000 ]* A ld, A H ld,
```
Comment (davazp) [inline lib/core.fs]: The order in memory of the register is usually the opposite, `L` first, then `H`, that will make easier to interop with Forth.

Comment (davazp) [inline lib/core.fs]: Also, perhaps `here` and `allot` are a bit too high level.

Couldn't we just expose `dp and dp!` words to set the dictionary pointer? then allot and here are easily defined in Forth.

Comment (Lapixx) [reply lib/core.fs]: That makes sense, but it is implemented this way around for `!` and `@` at the moment, so I’ll make sure change them everywhere.

Comment (Lapixx) [reply lib/core.fs]: We can do that as well, but I noticed that for example gforth does not seem to expose the `dp` directly. Not sure if there was some reasoning behind that.

Comment (davazp) [reply lib/core.fs]: well, I mean, exposing them from the kernel, but defining the higher level ones in the lib/ in Forth. But anyway it is probably more efficient in asm so it's fine.

**Pull request merged**